java/opendcs/src/main/java/decodes/tsdb/algo/WeightedAverage.java

```diff
@@ -0,0 +1,110 @@
+package decodes.tsdb.algo;
+
+import decodes.tsdb.DbCompException;
+import ilex.var.NamedVariable;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
+
+@Algorithm(description = "Calculate Weighted Average values (eg Water Temperature) using: sum(input1n*input2n) / total2")
+public class WeightedAverage
+	extends AW_AlgorithmBase
+{
+	@Input
+	public double weightTotal;
+	@Input
+	public double input1;
+	@Input
+	public double input2;
+	@Input
+	public double input3;
+	@Input
+	public double input4;
+	@Input
+	public double input5;
+	@Input
+	public double input6;
+	@Input
+	public double input7;
+	@Input
+	public double input8;
+	@Input
+	public double weight1;
```

Comment:
Are these time series?

Reply:
Yes, this is conceptual, and these will need to be altered to timeseries references. Not sure what the original algorithm is doing here.

Reply:
I've tested this and it's functioning. I'm going to add a bit more logging, and finalize the PR for review.